src/server/scripts/Kalimdor/Firelands/instance_firelands.cpp

```diff
@@ -40,8 +73,133 @@ class instance_firelands : public InstanceMapScript
                         // Cannot directly start attacking here as the creature is not yet on map
                         creature->m_Events.AddEvent(new DelayedAttackStartEvent(creature), creature->m_Events.CalculateTime(500));
                         break;
+                    case NPC_SULFURAS_SMASH_TRIGGER:
+                    case NPC_SULFURAS:
+                    case NPC_MOLTEN_ELEMENTAL:
+                    case NPC_BLAZING_HEAT:
+                        if (Creature* ragnaros = GetCreature(DATA_RAGNAROS))
+                            if (ragnaros->IsAIEnabled)
+                                ragnaros->AI()->JustSummoned(creature);
+                        break;
+                    // Ragnaros trash manual linked respawn
+                    case NPC_LAVA_WIELDER:
+                    case NPC_MOLTEN_SPEWER:
+                    case NPC_MOLTEN_ERUPTER:
+                        if (GetBossState(DATA_RAGNAROS) != DONE)
+                            break;
+
+                        creature->SetRespawnTime(WEEK);
+                        creature->DisappearAndDie();
+                        break;
+                    default:
+                        break;
+                }
+
+                InstanceScript::OnCreatureCreate(creature);
+            }
+
+            void OnGameObjectCreate(GameObject* go) override
+            {
+                switch(go->GetEntry())
+                {
+                    case GO_RAGNAROS_DOOR:
+                        AddDoor(go, true);
+                        break;
+                    case GO_RAGNAROS_PLATFORM:
+                        if (GetBossState(DATA_RAGNAROS) == DONE && instance->IsHeroic())
+                            go->SetDestructibleState(GO_DESTRUCTIBLE_DESTROYED);
+                        break;
+                    default:
+                        break;
                 }
+
+                InstanceScript::OnGameObjectCreate(go);
             }
+
+            void OnGameObjectRemove(GameObject* go) override
+            {
+                switch (go->GetEntry())
+                {
+                case GO_RAGNAROS_DOOR:
+                    AddDoor(go, false);
+                    break;
+                default:
+                    break;
+                }
+
+                InstanceScript::OnGameObjectRemove(go);
+            }
+
+            uint32 GetData(uint32 type) const override
+            {
+                switch (type)
+                {
+                    case DATA_RAGNAROS_FIRST_EMERGE:
+                        return _ragnarosFirstEmerge ? TRUE : FALSE;
```

Comment:
Uppercase `TRUE : FALSE` works fine in VS2015, but Clang 3.5.0 does not share that view.